x-pack/plugins/canvas/canvas_plugin_src/renderers/advanced_filter/component/index.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { compose, withState } from 'recompose';
+import { AdvancedFilter as Component, Props as ComponentProps } from './advanced_filter';
+
+export interface Props {
+  value?: string;
+  commit: (value: string) => void;
+}
+
+export const AdvancedFilter = compose<ComponentProps, Props>(
```

Comment:
I haven't used `compose` before, but I was using `connect` over in dashboard code.  If you have experience in both I'd be interested to hear the differences.  Also wondering about your thoughts on naming.  Over in dashboard I was doing something like `dashboard_panel_container.ts` is the wrapper and `dashboard_panel.ts` is the inner component.  It's not the best set up, and there was some back and forth on it (don't like leaking implementation details to users - what do they care if it's a wrapper or the actual thing, the "container" suffix means nothing to them).  Actually at some point we did something similar to this, so externally the component name looks the same as the internal name:

`export DashboardPanelContainer as DashboardPanel`; in the `index` file.

But, we migrated back to the first version when it became clear this was caused confusion when devs not familiar with the code were searching for the component being used - it's not obvious which is which.

Not something that needs to be solved in this PR but I'd be interested to hear your opinions on the matter and maybe we can come up with a "best practice" across the board for this.  Not too many users of redux at this point so probably wouldn't be too tough to align. (assuming this is even a redux thing and similar to the `connect` stuff being done in dashboard code).

Reply:
> I haven't used compose before, but I was using connect over in dashboard code. If you have experience in both I'd be interested to hear the differences.

`compose` and `connect` are not the same thing. `connect` is an HoC used with redux to inject state values as props. `compose` is a functional concept where each argument is executed and passed to the next. It's an easy way to tack on multiple HoC's. 

```js
// instead of writing this (warning, i might have the function order backwards here... i mix that up a lot)
connect(...)(withState(...)(withRouter(...)(Component)))

// you can write this (functions are called from left to right, no need to think about the order)
compose(
  connect(...),
  withState(...),
  withRouter(...)
)(Component)
```

In this specific instance, there's only one HoC, so compose isn't needed.